### PR TITLE
Feature to get all endpoints of an api

### PR DESCRIPTION
**Pull request by alahirajeffrey**

### CODE REVIEW

File: src/endpoints/endpoints.service.ts [hunk 1]
```diff
@@ -1,4 +1,41 @@
...
+export class EndpointsService {
+    constructor(
+        @InjectRepository(Endpoint)
+        private readonly endpointRepo : Repository<Endpoint>
+    ){}
+
+    async getAllApiEndpoints(dto: FindApiEndpoints){
+        try {
+            //check if api exists in Endpoint table
+            const apiExists = await this.endpointRepo.find({where : { apiId : dto.apiId }})
```
Comment (Oladimejiibitoye) [inline src/endpoints/endpoints.service.ts:19]: why are you creating a dto for the api_id?  You should get the id as a params, dto should not be required

File: src/endpoints/endpoints.controller.ts [hunk 2]
```diff
@@ -1,4 +1,18 @@
...
 
+@ApiTags('endpoints')
 @Controller('endpoints')
-export class EndpointsController {}
+export class EndpointsController {
+    constructor(
+        private readonly endpointsService : EndpointsService
+    ){}
+
+    @Get('api:apiId')
```
Comment (Oladimejiibitoye) [inline src/endpoints/endpoints.controller.ts:13]: I do not understand the construction of the route '('api:apiId')'

Comment (alahirajeffrey) [reply src/endpoints/endpoints.controller.ts:13]: it goes /categories/api:apiId

There might be an endpoint to get a category by id. 

The route above is there to differentiate and make it less ambiguous.

File: src/endpoints/endpoints.service.ts (hunk 1 above)
Comment (alahirajeffrey) [reply src/endpoints/endpoints.service.ts]: Done. Thanks

File: src/endpoints/endpoints.controller.ts (hunk 2 above)
Comment (TijanAyo) [reply src/endpoints/endpoints.controller.ts:12]: is it not supposed to go to **/categories/api/:apiID** the routing process is somewhat confusing

Comment (alahirajeffrey) [reply src/endpoints/endpoints.controller.ts:12]: the endpoint gets all the apis in an endpoint. 

The route is /endpoints/api/:apiId

**Review by johnayinde — APPROVED** [review]

**Pull request merged**